sympy/matrices/expressions/matexpr.py

```diff
@@ -1041,6 +1041,10 @@ def __new__(cls, m, n):
         cls._check_dim(m)
         cls._check_dim(n)
 
+        condition = Eq(m, 1) & Eq(n, 1)
+        if condition == True:
+            return Identity(1)
```

Comment:
I think that it should follow the design of `evaluate` keyword as done in matadd and matmul, even if it looks trivial.

Reply:
With a default of `False` as in `MatAdd` and `MatMul`? Then `OneMatrix(1, 1) == Identity(1)` wouldn't work anymore. When would the evaluation end up happening then?

Reply:
I think that matrix expressions follow up the design not to canonicalize even if it is cheap.
You'll notice that stuff like transpose, inverse towering up even if you build them on top of, but you should rather call `.doit` manually to make them structurally equal